packages/abi-coder/src/coders/byte.ts

```diff
@@ -54,9 +54,18 @@ export class ByteCoder extends Coder<number[], Uint8Array> {
   }
 
   decode(data: Uint8Array, offset: number): [Uint8Array, number] {
+    if (data.length < BASE_VECTOR_OFFSET) {
+      this.throwError(ErrorCode.DECODE_ERROR, 'Invalid byte data size.');
+    }
+
     const len = data.slice(16, 24);
     const length = bn(new U64Coder().decode(len, 0)[0]).toNumber();
-    const byteData = data.slice(BASE_VECTOR_OFFSET, BASE_VECTOR_OFFSET + length * 8);
+    const byteDataLength = length;
```

Comment:
@camsjams pretty sure we don't want to be multiplying by `WORD_SIZE` here. When length is initially encoded it is just the array length. Would have been fine as we were slicing from the correct point, but to a point larger than the size of the byte array. And as heap types go at the end of the bytecode didn't make a difference to integration tests. But potential for overflow issues.